src/components/safe-apps/SafeAppLandingPage/AppActions.tsx

```diff
@@ -1,38 +1,59 @@
-import type { ChainInfo } from '@gnosis.pm/safe-react-gateway-sdk'
-import { useState } from 'react'
+import type { ChainInfo, SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+import { useEffect, useMemo, useState } from 'react'
 import type { ConnectedWallet } from '@/hooks/wallets/useOnboard'
 import { useAppSelector } from '@/store'
-import { selectAddressBookByChain } from '@/store/addressBookSlice'
+import { selectAllAddressBooks } from '@/store/addressBookSlice'
+import { selectChains } from '@/store/chainsSlice'
 import useLastSafe from '@/hooks/useLastSafe'
 import { parsePrefixedAddress } from '@/utils/addresses'
 import SafeIcon from '@/components/common/SafeIcon'
 import EthHashInfo from '@/components/common/EthHashInfo'
 import { AppRoutes } from '@/config/routes'
+import useOwnedSafes from '@/hooks/useOwnedSafes'
 import { CTA_BUTTON_WIDTH, CTA_HEIGHT } from '@/components/safe-apps/SafeAppLandingPage/constants'
 
 type Props = {
   appUrl: string
   wallet: ConnectedWallet | null
   onConnectWallet: () => Promise<void>
-  safes: string[]
   chain: ChainInfo
+  app: SafeAppData
 }
 
-const AppActions = ({ wallet, onConnectWallet, safes, chain, appUrl }: Props): React.ReactElement => {
+type CompatibleSafesType = { address: string; chainId: string; shortName?: string }
+
+const AppActions = ({ wallet, onConnectWallet, chain, appUrl, app }: Props): React.ReactElement => {
   const lastUsedSafe = useLastSafe()
-  const lastUsedSafeAddress =
-    lastUsedSafe && safes.includes(lastUsedSafe) ? parsePrefixedAddress(lastUsedSafe).address : ''
-  const [safeToUse, setSafeToUse] = useState<string | undefined>(lastUsedSafeAddress || undefined)
-  const addressBook = useAppSelector((state) => selectAddressBookByChain(state, chain.chainId))
+  const ownedSafes = useOwnedSafes()
+  const addressBook = useAppSelector(selectAllAddressBooks)
+  const chains = useAppSelector(selectChains)
+  const compatibleChains = app.chainIds
+
+  const compatibleSafes = useMemo(
+    () => getCompatibleSafes(ownedSafes, compatibleChains, chains.data),
+    [ownedSafes, compatibleChains, chains.data],
+  )
+
+  const [safeToUse, setSafeToUse] = useState<CompatibleSafesType>()
+
+  useEffect(() => {
+    const selectedSafe = getDefaultSafe(compatibleSafes, chain.chainId, lastUsedSafe)
+    if (selectedSafe) {
+      setSafeToUse(selectedSafe)
+    }
+  }, [compatibleSafes, chain.chainId, lastUsedSafe])
+
   const hasWallet = !!wallet
-  const hasSafes = safes.length > 0
+  const hasSafes = compatibleSafes.length > 0
   const shouldCreateSafe = hasWallet && !hasSafes
 
   let button: React.ReactNode
   switch (true) {
     case hasWallet && hasSafes:
-      const href = `${AppRoutes.apps}?appUrl=${encodeURIComponent(appUrl)}&safe=${chain.shortName}:${safeToUse}`
+      const href = `${AppRoutes.apps}?appUrl=${encodeURIComponent(appUrl)}&safe=${safeToUse?.shortName}:${
```

Comment:
We decided to avoid string URLs and use URL Objects as much as possible. Could you please convert this one?

Reply:
It seems that [`new URL(..)`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) requires an absolute url. But in our case we have a relative one:

![Captura de pantalla 2022-10-04 a las 15 47 04](https://user-images.githubusercontent.com/26763673/193836401-9b99cc92-2510-475d-889f-cfcca51b2492.png)


For internal redirection we are using string URLs:

![Captura de pantalla 2022-10-04 a las 15 42 49](https://user-images.githubusercontent.com/26763673/193835641-80982121-2de4-4478-a3ab-74891c56b74c.png)
![Captura de pantalla 2022-10-04 a las 15 43 25](https://user-images.githubusercontent.com/26763673/193835645-0e3244b2-cb3e-402c-a443-2e00baf94b97.png)

Reply:
It takes simple objects, not necessarily URLObject objects, like so:
```
<Link href={{
  path: '/welcome',
  { query: { chain: 'rin' } }
}} />
```

Reply:
See https://github.com/safe-global/web-core/pull/783